quic/s2n-quic-transport/src/dc/manager.rs

```diff
@@ -0,0 +1,209 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    contexts::WriteContext,
+    endpoint,
+    sync::{flag, flag::Writer},
+};
+use s2n_quic_core::{
+    ack,
+    crypto::tls,
+    dc,
+    dc::{Endpoint, Path},
+    ensure, event,
+    event::builder::{DcState, DcStateChanged},
+    frame::DcStatelessResetTokens,
+    packet::number::PacketNumber,
+    state::is,
+    stateless_reset, transmission,
+    transmission::interest::Query,
+};
+
+/// Manages transmission and receipt of `DC_STATELESS_RESET_TOKENS` and
+/// notifications to the dc::Endpoint Path
+pub struct Manager<Config: endpoint::Config> {
+    path: Option<<<Config as endpoint::Config>::DcEndpoint as Endpoint>::Path>,
+    version: Option<dc::Version>,
+    state: State,
+}
+
+type Flag = flag::Flag<DcStatelessResetTokenWriter>;
+
+#[derive(Debug, Default)]
+enum State {
+    #[default]
+    Init,
+    PathSecretsReady(Flag),
```

Comment:
Might be a bit simpler to just put the `Flag` on the `Manager` so you don't have to match on it for interests. You also get to use the state transitions macro which can make it a bit clearer which transitions are valid.

Reply:
the only thing that is a little weird with this approach is that the dc provider doesn't have the stateless reset tokens ready when the dc Manager is initialized, so the flag would just be Flag::default() and then it would get overwritten with the actual flag later on when path secrets are ready. Not a big deal though

Reply:
Yeah default should be fine

Reply:
I ended up adding a new state `ServerTokensSent`, that the Server transitions to after it has received tokens from the client. I now have the server's flag only activating once it has received tokens from the Client, so that we don't end up in a situation where the Client didn't send tokens or they got lost, and the Server sent tokens and had them acked, and thus thinking it was done.